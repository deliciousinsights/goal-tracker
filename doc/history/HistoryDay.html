<!DOCTYPE html><html lang="en"><head><title>history/HistoryDay</title></head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0"><meta name="groc-relative-root" content="../"><meta name="groc-document-path" content="history/HistoryDay"><meta name="groc-project-path" content="src/history/HistoryDay.js"><link rel="stylesheet" type="text/css" media="all" href="../assets/style.css"><script type="text/javascript" src="../assets/behavior.js"></script><body><div id="meta"><div class="file-path">src/history/HistoryDay.js</div></div><div id="document"><div class="segment"><div class="comments "><div class="wrapper"><h1 id="journe-dhistorique">Journée d’historique</h1>
<p>Section de l&#39;écran d’historique, dédiée à un jour précis.</p></div></div><div class="code"><div class="wrapper"><span class="hljs-keyword">import</span> Divider <span class="hljs-keyword">from</span> <span class="hljs-string">'@mui/material/Divider'</span>
<span class="hljs-keyword">import</span> List <span class="hljs-keyword">from</span> <span class="hljs-string">'@mui/material/List'</span>
<span class="hljs-keyword">import</span> Typography <span class="hljs-keyword">from</span> <span class="hljs-string">'@mui/material/Typography'</span>

<span class="hljs-keyword">import</span> {
  arrayOf,
  GoalPropType,
  HistoryDayStatsPropType,
} <span class="hljs-keyword">from</span> <span class="hljs-string">'../shared/prop-types'</span>
<span class="hljs-keyword">import</span> { formatDate } <span class="hljs-keyword">from</span> <span class="hljs-string">'../lib/helpers'</span>
<span class="hljs-keyword">import</span> HistoryDayGoal <span class="hljs-keyword">from</span> <span class="hljs-string">'./HistoryDayGoal'</span>

export default function HistoryDay({ goals, stats: { date, progresses } }) {
  return (
    &lt;div&gt;
      &lt;Divider /&gt;
      &lt;List&gt;
        &lt;Typography variant=<span class="hljs-string">'subtitle1'</span>&gt;{formatDate(date)}&lt;/Typography&gt;
        {goals.map((goal) =&gt; {
          const goalStats = progresses[goal.id]
          if (goalStats) {
            return (
              &lt;HistoryDayGoal key={goal.id} goal={goal} stats={goalStats} /&gt;
            )
          }
          return null
        })}
      &lt;/List&gt;
    &lt;/div&gt;
  )
}</div></div></div><div class="segment"><div class="comments "><div class="wrapper"><p>Comme toujours, on définit les propriétés attendues/autorisées pour
validation.</p></div></div><div class="code"><div class="wrapper">HistoryDay.propTypes = {
  goals: arrayOf(GoalPropType).isRequired,
  stats: HistoryDayStatsPropType.isRequired,
}</div></div></div></div></body></html>