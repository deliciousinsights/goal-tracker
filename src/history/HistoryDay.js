// Journée d’historique
// ====================

// Section de l'écran d’historique, dédiée à un jour précis.

import Divider from '@mui/material/Divider'
import List from '@mui/material/List'
import Typography from '@mui/material/Typography'

import {
  arrayOf,
  GoalPropType,
  HistoryDayStatsPropType,
} from '../shared/prop-types'
import { formatDate } from '../lib/helpers'
import HistoryDayGoal from './HistoryDayGoal'

export default function HistoryDay({ goals, stats: { date, progresses } }) {
  return (
    <div>
      <Divider />
      <List>
        <Typography variant='subtitle1'>{formatDate(date)}</Typography>
        {goals.map((goal) => {
          const goalStats = progresses[goal.id]
          if (goalStats) {
            return (
              <HistoryDayGoal key={goal.id} goal={goal} stats={goalStats} />
            )
          }
          return null
        })}
      </List>
    </div>
  )
}

// Comme toujours, on définit les propriétés attendues/autorisées pour
// validation.
HistoryDay.propTypes = {
  goals: arrayOf(GoalPropType).isRequired,
  stats: HistoryDayStatsPropType.isRequired,
}
