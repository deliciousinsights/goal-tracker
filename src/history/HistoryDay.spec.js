import { render } from '@testing-library/react'

import HistoryDay from './HistoryDay'

describe('<HistoryDay />', () => {
  const goals = [
    {
      id: '5bf57a79890a6e2c11ec9665',
      name: 'Demo goal 1',
      target: 5,
      units: 'wombats',
    },
    {
      id: '5bf57a79890a6e2c11ec9666',
      name: 'Demo goal 2',
      target: 10,
      units: 'weazels',
    },
  ]

  it('should otherwise match the expected snapshot', () => {
    const stats = { date: '2017-03-23', progresses: { 0: [1, 5] } }
    const { container } = render(<HistoryDay goals={goals} stats={stats} />)

    expect(container).toMatchSnapshot()
  })
})
