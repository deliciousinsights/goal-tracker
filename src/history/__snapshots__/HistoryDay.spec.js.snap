// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoryDay /> should otherwise match the expected snapshot 1`] = `
<div>
  <div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
      >
        jeudi 23 mars 2017
      </h6>
    </ul>
  </div>
</div>
`;
